source/simulator/solver.cc

```diff
@@ -919,7 +926,8 @@ namespace aspect
     if (parameters.include_melt_transport)
       melt_handler->compute_melt_variables(solution);
 
-    return initial_residual;
+    return std::pair<double,double>(final_linear_residual,
+                                    initial_nonlinear_residual);
```

Comment:
looking at the documentation in simulator.h and the code in solver_schemes, I think you mend to do this the other way around.